internal/gitinterface/repository.go

```diff
@@ -0,0 +1,143 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package gitinterface
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/go-git/go-git/v5"
+	"github.com/jonboulle/clockwork"
+)
+
+const (
+	binary           = "git"
+	committerTimeKey = "GIT_COMMITTER_DATE"
+	authorTimeKey    = "GIT_AUTHOR_DATE"
+)
+
+// Repository is a lightweight wrapper around a Git repository. It stores the
+// location of the repository's GIT_DIR.
+type Repository struct {
+	gitDirPath string
+	clock      clockwork.Clock
+}
+
+// GetGoGitRepository returns the go-git representation of a repository. We use
+// this in certain signing and verifying workflows.
+func (r *Repository) GetGoGitRepository() (*git.Repository, error) {
+	return git.PlainOpenWithOptions(r.gitDirPath, &git.PlainOpenOptions{DetectDotGit: true})
+}
+
+// GetGitDir returns the GIT_DIR path for the repository.
+func (r *Repository) GetGitDir() string {
+	return r.gitDirPath
+}
+
+// LoadRepository returns a Repository instance using the current working
+// directory. It also inspects the PATH to ensure Git is installed.
+func LoadRepository() (*Repository, error) {
+	_, err := exec.LookPath(binary)
+	if err != nil {
+		return nil, fmt.Errorf("unable to find Git binary, is Git installed?")
+	}
+
+	repo := &Repository{clock: clockwork.NewRealClock()}
+	envVar := os.Getenv("GIT_DIR")
```

Comment:
Is this a git convention or our own thing?

Reply:
This is sometimes set by Git: https://git-scm.com/book/en/v2/Git-Internals-Environment-Variables